staging/src/k8s.io/client-go/1.5/rest/config.go

```diff
@@ -109,6 +110,9 @@ type Config struct {
 	// Rate limiter for limiting connections to the master from this client. If present overwrites QPS/Burst
 	RateLimiter flowcontrol.RateLimiter
 
+	// The maximum length of time to wait before giving up on a server request. A value of zero means no timeout.
+	Timeout time.Duration
+
```

Comment:
What about actually setting this field like you do in the core k8s code? I'd expect this patch looks identical to the other `config.go` file.